src/components/WwwFrame/TheHeader.vue

```diff
@@ -597,6 +624,24 @@ export default {
 			}
 		}
 	},
+	created() {
+		if (!this.isVisitor) {
+			try {
+				let userData = {};
+				userData = this.apollo.readQuery({
```

Comment:
This will throw an error if it wasn't called in the `preFetch`. Generally you don't want to create code that has _expected_ exceptions. Instead the `preFetch` calling should match the `created`. For example, this should never throw an exception unless there is some kind of expected network error.

Reply:
My suggestion here it to perform the same cookie check you do in the prefetch as a gate to using readQuery. If either are undefined, you would have prefetched the query.

Reply:
The edge case was when a user has the cookies, then takes an action that would change the value saved to the cookie. Can we ignore that use case? If so, it simplifies this logic quite a bit.

Reply:
The actions that would change these cookies are confined to the new "thanks" contexts. This "global" query will only run when the user is logged in and does not have both cookies already set.

Reply:
Sounds like we aren't worried about changing the cookie value, that simplifies things a little. CC: @edwinf-barreram